src/main/java/uk/gov/companieshouse/web/accounts/CompanyAccountsWebApplication.java

```diff
@@ -1,12 +1,28 @@
 package uk.gov.companieshouse.web.accounts;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import uk.gov.companieshouse.web.accounts.interceptor.UserDetailsInterceptor;
 
 @SpringBootApplication
-public class CompanyAccountsWebApplication {
+public class CompanyAccountsWebApplication implements WebMvcConfigurer {
+
+    UserDetailsInterceptor userDetailsInterceptor;
+
+    @Autowired
+    public CompanyAccountsWebApplication(UserDetailsInterceptor userDetailsInterceptor) {
+        this.userDetailsInterceptor = userDetailsInterceptor;
+    }
 
     public static void main(String[] args) {
         SpringApplication.run(CompanyAccountsWebApplication.class, args);
     }
+
+    @Override
+    public void addInterceptors(InterceptorRegistry registry) {
+        registry.addInterceptor(userDetailsInterceptor);
```

Comment:
does this automatically add this interceptor to every route? I've seen this done with `.addPathPatterns("/**")`

https://github.com/companieshouse/transactions.api.ch.gov.uk/blob/develop/src/main/java/uk/gov/companieshouse/api/transactions/TransactionsApplication.java#L82

Reply:
It does work for all routes, but `addPathPatterns` can be added to make it more explicit. Typically `addPathPatterns` should be used to exclude certain routes.